src/utils/storageFacade.utils.ts

```diff
@@ -1,122 +1,355 @@
+ */
+export async function readVulnsByPkg(packageRef: string, sessionId: number) {
+  const { data, error } = await supabase
+    .from('vulnerabilities')
+    .select('*,junction!inner(packageid,packages!inner(package_ref))')
+    .eq('junction.packages.package_ref', packageRef);
+  let cves: Vulnerability[] = [];
+
+  if (error) {
+    logger.error(error.message);
+  } else if (data) {
+    for (let v of data) {
+      cves.push(
+        //map database result to Vulnerability object
+        {
+          cveId: v.cveidstring,
+          packageRef: v.junction[0].packages.package_ref,
+          impact: v.impact,
+          likelihood: v.likelihood,
+          risk: v.risk,
+          cvss2: v.cvss_vector,
+        }
+      );
+    }
+  }
+  return cves;
+}
+
+/**
+ * Write Package, updates if entry exists in database, inserts otherwise
+ * @param cve to write
+ * @param sessionId associated with one user
+ * @returns supabase write status
+ */
+export async function writeVuln(cve: Vulnerability, sessionId: number) {
+  let { data, error } = await supabase
+    .from('vulnerabilities')
+    .select('*')
+    .eq('cveidstring', cve.cveId);
+
+  if (error) {
+    logger.error(error.message);
+    return error;
+  } else if (data) {
+    if (data.length == 0) {
+      //INSERT
+      return insertVuln(cve, sessionId);
+    } else {
+      //UPDATE
+      return updateVuln(cve, sessionId);
+    }
+  }
+  return 400;
+}
+
+/******************PACKAGE HELPERS**********************/
+//Insert package by package id
+async function insertPackage(pkg: Package, sessionId: number) {
+  let { status, error } = await supabase.from('packages').insert({
+    //append necessary housekeeping info for database
+    name: pkg.name,
+    packageversion: pkg.version,
+    consrisk: pkg.consRisk,
+    impact: pkg.impact,
+    likelihood: pkg.likelihood,
+    highestrisk: pkg.highestRisk,
+    purl: pkg.purl,
+    cpename: pkg.cpeName,
+    sessionid: sessionId,
+    package_ref: pkg.ref,
+  });
+  if (error) {
+    logger.error(error.message);
+  }
+  return status;
+}
+
+async function updatePackage(pkg: Package, sessionId: number) {
+  let { status, error } = await supabase
+    .from('packages')
+    .update({
+      //append necessary housekeeping info for database
+      name: pkg.name,
+      packageversion: pkg.version,
+      consrisk: pkg.consRisk,
+      impact: pkg.impact,
+      likelihood: pkg.likelihood,
+      highestrisk: pkg.highestRisk,
+      purl: pkg.purl,
+      cpename: pkg.cpeName,
+    })
+    .eq('sessionid', sessionId)
+    .eq('package_ref', pkg.ref);
+  if (error) {
+    logger.error(error.message);
+  }
+  return status;
+}
+
 //Function #4: Write Single or Many Request (Package)
-export async function writePackage(DBPackage:any){
-  let {status} = await supabase
-  .from('packages')
-  .insert(DBPackage);
-  return status; 
+async function writePackageByVulType(DBPackage: any) {
```

Comment:
recall that DBPackage is no longer an interface in types.utils because it was deleted - should either rewrite this function or add DBPackage back

Reply:
I will update - thanks